cmd/backup/storages/local.go

```diff
@@ -0,0 +1,118 @@
+package storages
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	t "github.com/offen/docker-volume-backup/cmd/backup/types"
+	u "github.com/offen/docker-volume-backup/cmd/backup/utilities"
+	"github.com/sirupsen/logrus"
+)
+
+type LocalStorage struct {
+	*GenericStorage
+}
+
+// Specific init procedure for the local storage provider.
+func InitLocal(c *t.Config, l *logrus.Logger) *LocalStorage {
+	a := &GenericStorage{&LocalStorage{}, l, c}
+	r := &LocalStorage{a}
+	return r
+}
+
+// Specific copy function for the local storage provider.
+func (stg *LocalStorage) copy(file string) error {
+	_, name := path.Split(file)
+
+	if err := u.CopyFile(file, path.Join(stg.config.BackupArchive, name)); err != nil {
+		return fmt.Errorf("copyBackup: error copying file to local archive: %w", err)
+	}
+	stg.logger.Infof("Stored copy of backup `%s` in local archive `%s`.", file, stg.config.BackupArchive)
+
+	if stg.config.BackupLatestSymlink != "" {
```

Comment:
I wonder if it would be better to be more explicit about the symlinking behavior. Should this be a separate step and not be buried in `copy`? I can see pros and cons for both options so I'm not sure right now.

Reply:
Well, since we are already taking the code apart, I would vote for making it a separate function. Probably private and used by `copy`, but it's still a distinct enough step imo.